packages/fcl/src/wallet-provider-spec/user-signature.md

```diff
@@ -0,0 +1,266 @@
+# User Signature
+
+## Status
+
+- **Last Updated:** June 1st 2021
+- **Stable:** Yes
+- **Risk of Breaking Change:** Medium
+- **Compatibility:** `>= @onflow/fcl@0.0.70`
+
+# Overview and Introduction
+
+**Personally sign data via FCL Compatible Wallets**
+
+**FCL** now incldues **`signUserMessage()`** which allows for the sending of unencrypted message data to a connected wallet provider or service to be signed with a user's private key. 
+
+An application or service can verify a signature against a user's public key on the **Flow Blockchain** providing proof a user controls the account's private key.   
+
+**Use Cases**
+
+- **Authentication**: Cryptographically verify the ownership of a **Flow** account by signing a piece of data using a private key
+- **Improved Application Login**
+  - **Increased security**: Arguably more secure than proof of ownership by email/password
+  - **Simplified UX**: No application password required
+  - **Increased privacy**: No email or third party authentication service needed
+- **Message Validation**: Assuring that a message sent or received has not been tampered with
+- **Multisig contracts**
+- **Decentralised exchanges**
+- **Meta transactions**
+
+
+# Config and Authentication
+
+As a prerequisite, **FCL** is configured to point to the Wallet Provider's Authentication Endpoint.
+
+> During development (and on mainnet) FCL can be configured to use the wallet directly by
+> setting the **Challenge Handshake Url** to the wallet providers **Authentication Endpoint**
```

Comment:
`challenge.handshake` is deprecated in favour of `discovery.wallet`

There are docs on configuration here: https://github.com/onflow/flow-js-sdk/blob/master/docs/configure-fcl.mdx#common-configuration-keys

Reply:
updated and included link to config docs